TestDump failure with some kernels on arm64

1. What version of Delve are you using (dlv version)?
v1.6.1, obtained by git and checkout of the tag
Also master @ a2b839990e21690e9e60a74c582c4acece6196d8

2. What version of Go are you using? (go version)?
go version go1.16.6 linux/arm64

3. What operating system and processor architecture are you using?
Ubuntu Hirsute on arm64
uname -a output:
Linux pi 5.11.0-1012-raspi 13-Ubuntu SMP PREEMPT Thu Jun 17 10:44:37 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux

4. What did you do?
go test ./... -run TestDump

5. What did you expect to see?
Passing tests :)

6. What did you see instead?
=== RUN   TestDump
    proc_test.go:5117: testing normal dump
    proc_test.go:5029: bad DumpState &proc.DumpState{Mutex:sync.Mutex{state:0, sema:0x0}, Dumping:false, AllDone:true, Canceled:false, DoneChan:(chan struct {})(nil), ThreadsDone:5, ThreadsTotal:5, MemDone:0x657b000, MemTotal:0x657d000, Err:error(nil)}
--- FAIL: TestDump (0.98s)

Greetings!

Ubuntu automated tests for delve are consistently failing on TestDump on arm64.
I suspect a kernel issue, given that on older kernels this failure doesn't occur.
https://bugs.launchpad.net/ubuntu/+source/delve/+bug/1938474

The test fails becaue MemDone != MemTotal.
The failing item is vvar.
err when attempting to read is
error(syscall.Errno) golang.org/x/sys/unix.EIO (5)

Notably however, this test passes in debian CI. (kernel 4.19)
Also, running on older series such as bionic/focal does pass.

vvar section from vm with kernel 5.4 (test pass):
ffff924a4000-ffff924a5000 r--p 00000000 00:00 0 [vvar]
Size: 4 kB
KernelPageSize: 4 kB
MMUPageSize: 4 kB
Rss: 0 kB
Pss: 0 kB
Shared_Clean: 0 kB
Shared_Dirty: 0 kB
Private_Clean: 0 kB
Private_Dirty: 0 kB
Referenced: 0 kB
Anonymous: 0 kB
LazyFree: 0 kB
AnonHugePages: 0 kB
ShmemPmdMapped: 0 kB
FilePmdMapped: 0 kB
Shared_Hugetlb: 0 kB
Private_Hugetlb: 0 kB
Swap: 0 kB
SwapPss: 0 kB
Locked: 0 kB
THPeligible: 0
VmFlags: rd mr de

vvar from failing case
ffffac144000-ffffac146000 r--p 00000000 00:00 0 [vvar]
Size: 8 kB
KernelPageSize: 4 kB
MMUPageSize: 4 kB
Rss: 0 kB
Pss: 0 kB
Shared_Clean: 0 kB
Shared_Dirty: 0 kB
Private_Clean: 0 kB
Private_Dirty: 0 kB
Referenced: 0 kB
Anonymous: 0 kB
LazyFree: 0 kB
AnonHugePages: 0 kB
ShmemPmdMapped: 0 kB
FilePmdMapped: 0 kB
Shared_Hugetlb: 0 kB
Private_Hugetlb: 0 kB
Swap: 0 kB
SwapPss: 0 kB
Locked: 0 kB
THPeligible: 0
VmFlags: rd mr pf de

Two noticible differences
1) 4kB vs 8kB size
2) Failing case adds the "pf" flag ("pure PFN range")

Filing here as I'm unsure where the issue lies or if some sort of workaround is desired.

--- a/pkg/proc/native/dump_linux.go
+++ b/pkg/proc/native/dump_linux.go
@@ -47,6 +47,9 @@ smapsLinesLoop:
 
 		for i := range vmflags {
 			switch vmflags[i] {
+			case "pf":
+				// pure PFN range, see https://github.com/go-delve/delve/issues/2630
+				continue smapsLinesLoop
 			case "dd":
 				// "don't dump"
 				continue smapsLinesLoop

